diff --git a/__pycache__/agent.cpython-37.pyc b/__pycache__/agent.cpython-37.pyc
index a898606..7e865c1 100644
Binary files a/__pycache__/agent.cpython-37.pyc and b/__pycache__/agent.cpython-37.pyc differ
diff --git a/__pycache__/dqn.cpython-37.pyc b/__pycache__/dqn.cpython-37.pyc
index 585a6da..6b515fd 100644
Binary files a/__pycache__/dqn.cpython-37.pyc and b/__pycache__/dqn.cpython-37.pyc differ
diff --git a/__pycache__/nash_dqn.cpython-37.pyc b/__pycache__/nash_dqn.cpython-37.pyc
index 77dc5fb..550c131 100644
Binary files a/__pycache__/nash_dqn.cpython-37.pyc and b/__pycache__/nash_dqn.cpython-37.pyc differ
diff --git a/common/__pycache__/args_parser.cpython-37.pyc b/common/__pycache__/args_parser.cpython-37.pyc
index 746c9ec..a496bdd 100644
Binary files a/common/__pycache__/args_parser.cpython-37.pyc and b/common/__pycache__/args_parser.cpython-37.pyc differ
diff --git a/common/__pycache__/networks.cpython-37.pyc b/common/__pycache__/networks.cpython-37.pyc
index b6e3211..dd72593 100644
Binary files a/common/__pycache__/networks.cpython-37.pyc and b/common/__pycache__/networks.cpython-37.pyc differ
diff --git a/common/__pycache__/nn_components.cpython-37.pyc b/common/__pycache__/nn_components.cpython-37.pyc
index 1a8e4b3..7c8b91d 100644
Binary files a/common/__pycache__/nn_components.cpython-37.pyc and b/common/__pycache__/nn_components.cpython-37.pyc differ
diff --git a/common/__pycache__/storage.cpython-37.pyc b/common/__pycache__/storage.cpython-37.pyc
index da5c63b..b539601 100644
Binary files a/common/__pycache__/storage.cpython-37.pyc and b/common/__pycache__/storage.cpython-37.pyc differ
diff --git a/common/__pycache__/wrappers.cpython-37.pyc b/common/__pycache__/wrappers.cpython-37.pyc
index 85a9d85..f0e80d1 100644
Binary files a/common/__pycache__/wrappers.cpython-37.pyc and b/common/__pycache__/wrappers.cpython-37.pyc differ
diff --git a/equilibrium_solver/__pycache__/__init__.cpython-37.pyc b/equilibrium_solver/__pycache__/__init__.cpython-37.pyc
index 74fa21a..50f54f1 100644
Binary files a/equilibrium_solver/__pycache__/__init__.cpython-37.pyc and b/equilibrium_solver/__pycache__/__init__.cpython-37.pyc differ
diff --git a/equilibrium_solver/__pycache__/eq_CVXPYsolver.cpython-37.pyc b/equilibrium_solver/__pycache__/eq_CVXPYsolver.cpython-37.pyc
index 4c2ec57..e689ef1 100644
Binary files a/equilibrium_solver/__pycache__/eq_CVXPYsolver.cpython-37.pyc and b/equilibrium_solver/__pycache__/eq_CVXPYsolver.cpython-37.pyc differ
diff --git a/equilibrium_solver/__pycache__/eq_ECOSsolver.cpython-37.pyc b/equilibrium_solver/__pycache__/eq_ECOSsolver.cpython-37.pyc
index 8a476d3..c3e39e2 100644
Binary files a/equilibrium_solver/__pycache__/eq_ECOSsolver.cpython-37.pyc and b/equilibrium_solver/__pycache__/eq_ECOSsolver.cpython-37.pyc differ
diff --git a/equilibrium_solver/__pycache__/eq_GUROBIsolver.cpython-37.pyc b/equilibrium_solver/__pycache__/eq_GUROBIsolver.cpython-37.pyc
index ef7eb67..d2a7144 100644
Binary files a/equilibrium_solver/__pycache__/eq_GUROBIsolver.cpython-37.pyc and b/equilibrium_solver/__pycache__/eq_GUROBIsolver.cpython-37.pyc differ
diff --git a/equilibrium_solver/__pycache__/eq_LPsolver.cpython-37.pyc b/equilibrium_solver/__pycache__/eq_LPsolver.cpython-37.pyc
index 1d7542f..2fcd49b 100644
Binary files a/equilibrium_solver/__pycache__/eq_LPsolver.cpython-37.pyc and b/equilibrium_solver/__pycache__/eq_LPsolver.cpython-37.pyc differ
diff --git a/equilibrium_solver/__pycache__/eq_MWUsolver.cpython-37.pyc b/equilibrium_solver/__pycache__/eq_MWUsolver.cpython-37.pyc
index 24ea6d4..b8153f7 100644
Binary files a/equilibrium_solver/__pycache__/eq_MWUsolver.cpython-37.pyc and b/equilibrium_solver/__pycache__/eq_MWUsolver.cpython-37.pyc differ
diff --git a/equilibrium_solver/__pycache__/eq_NASHPYsolver.cpython-37.pyc b/equilibrium_solver/__pycache__/eq_NASHPYsolver.cpython-37.pyc
index d77f2e0..84ba6c7 100644
Binary files a/equilibrium_solver/__pycache__/eq_NASHPYsolver.cpython-37.pyc and b/equilibrium_solver/__pycache__/eq_NASHPYsolver.cpython-37.pyc differ
diff --git a/equilibrium_solver/__pycache__/gamegenerator.cpython-37.pyc b/equilibrium_solver/__pycache__/gamegenerator.cpython-37.pyc
index 03b5c5c..f78e71a 100644
Binary files a/equilibrium_solver/__pycache__/gamegenerator.cpython-37.pyc and b/equilibrium_solver/__pycache__/gamegenerator.cpython-37.pyc differ
diff --git a/equilibrium_solver/__pycache__/lamke_howson_lex.cpython-37.pyc b/equilibrium_solver/__pycache__/lamke_howson_lex.cpython-37.pyc
index 3d2fa24..233db5b 100644
Binary files a/equilibrium_solver/__pycache__/lamke_howson_lex.cpython-37.pyc and b/equilibrium_solver/__pycache__/lamke_howson_lex.cpython-37.pyc differ
diff --git a/launch.sh b/launch.sh
old mode 100644
new mode 100755
diff --git a/wandb/debug-internal.log b/wandb/debug-internal.log
index 0b7feb1..0e2d471 120000
--- a/wandb/debug-internal.log
+++ b/wandb/debug-internal.log
@@ -1 +1 @@
-run-20220725_231135-2yxgsp5e/logs/debug-internal.log
\ No newline at end of file
+run-20220725_233854-vsl5h2zn/logs/debug-internal.log
\ No newline at end of file
diff --git a/wandb/debug.log b/wandb/debug.log
index b837fd7..7d59b76 120000
--- a/wandb/debug.log
+++ b/wandb/debug.log
@@ -1 +1 @@
-run-20220725_231135-2yxgsp5e/logs/debug.log
\ No newline at end of file
+run-20220725_233854-vsl5h2zn/logs/debug.log
\ No newline at end of file
diff --git a/wandb/latest-run b/wandb/latest-run
index 9a8fcb7..8b85730 120000
--- a/wandb/latest-run
+++ b/wandb/latest-run
@@ -1 +1 @@
-run-20220725_231135-2yxgsp5e
\ No newline at end of file
+run-20220725_233854-vsl5h2zn
\ No newline at end of file
